internal/clients/init.go

```diff
@@ -181,22 +162,22 @@ func checkConsulAvailable() bool {
 }
 
 func initializeClients() {
-	isRegistry := common.UseRegistry
 	var waitGroup sync.WaitGroup
 	waitGroup.Add(clientCount)
 
-	endpoint := &endpoint.Endpoint{RegistryClient: config.RegistryClient, WG: &waitGroup}
-
+	endpoint := &endpoint.Endpoint{RegistryClient: common.RegistryClient, WG: &waitGroup}
 	metaAddr := common.CurrentConfig.Clients[common.ClientMetadata].Url()
 	dataAddr := common.CurrentConfig.Clients[common.ClientData].Url()
+	isRegistry := common.RegistryClient != nil
 
 	params := types.EndpointParams{
 		UseRegistry: isRegistry,
 		Interval:    15,
 	}
 
 	// initialize Core Metadata clients
-	params.ServiceKey = common.CurrentConfig.Clients[common.ClientMetadata].Name
+	// hard-coded here, should be fixed in PR#447
+	params.ServiceKey = "edgex-core-metadata"
```

Comment:
These constants are available from go-mod-core-contracts. Please use them and then PR#447 doesn't have to fix this.
https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/constants.go#L61

Reply:
This is going to be immediately overwritten anyway so I wouldn't worry about it.